src/util.cc

```diff
@@ -630,3 +632,46 @@ bool Truncate(const string& path, size_t size, string* err) {
   }
   return true;
 }
+
+bool ReplaceContent(const string& file_dst, const string& new_content,
+                    string* err) {
+#ifdef _WIN32
+  if (!DeleteFile(file_dst.c_str())) {
+    *err = GetLastErrorString();
+    return false;
+  }
+
+  if (rename(new_content.c_str(), file_dst.c_str()) < 0) {
```

Comment:
Probably this should be done via

https://docs.microsoft.com/en-us/windows/win32/fileio/deprecation-of-txf?redirectedfrom=MSDN
https://docs.microsoft.com/en-us/windows/win32/api/winbase/nf-winbase-replacefilea

to provide the same atomic replacement semantics as on posix, but also with ACL information copied over.

It also preserves the creation time, but i think updates the modification time like we would want it to.

This API's been available since Windows XP, as far as I can tell from the documentation page.

Reply:
But if you don't want to do that, the original file shouldn't be deleted or renamed until after the original file is on disk regardless, so the unlink of the original should happen last